### PR TITLE
[jwt] create jwt subproject

### PR DESCRIPTION
**Pull request by danking**
cc: @akotlar @jigold @cseed 

Is everyone OK with this? You can run `python setup.py install` to install this project.

I want to use this in the cookies batch PR.

### CODE REVIEW

File: hailjwt/hailjwt/hailjwt.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+            return None
+        return jwt.decode(
+            token, self.secret_key, algorithms=[JWTClient.__ALGORITHM])
+
+    def encode(self, payload):
+        return jwt.encode(
+            payload, self.secret_key, algorithm=JWTClient.__ALGORITHM)
+
+
+def get_domain(host):
```
Comment (cseed) [inline hailjwt/hailjwt/hailjwt.py]: What is `get_domain` for?

File: hailjwt/test/test_jwt.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import uuid
+import hailjwt as hj
+
+
+def test_round_trip():
```
Comment (cseed) [inline hailjwt/test/test_jwt.py]: We should also verify rejection (1) corrupted key, and (2) corrupted cipher text.

File: hailjwt/test/test_jwt.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    c = hj.JWTClient(uuid.uuid4().hex)
```
Comment (cseed) [inline hailjwt/test/test_jwt.py]: Set a good example: use Python `secrets` to generate cryptographically strong random keys: https://docs.python.org/3.6/library/secrets.html.

File: hailjwt/hailjwt/hailjwt.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import jwt
+
+
+class JWTClient:
+    __ALGORITHM = 'HS256'
+
+    def __init__(self, secret_key):
+        self.secret_key = secret_key
```
Comment (cseed) [inline hailjwt/hailjwt/hailjwt.py]: You should assert that the secret_key is at least 256-bits in length.

"A key of the same size as the hash output (for instance, 256 bits for "HS256") or larger MUST be used with this algorithm."

https://tools.ietf.org/html/rfc7518#section-3.2

I think you should have a class method to generate the secret key using the `secrets` package, see my other comment.

File: hailjwt/hailjwt/hailjwt.py (hunk 1 above)
Comment (danking) [reply hailjwt/hailjwt/hailjwt.py]: I think @akotlar uses it to determine the root domain for the cookies

File: hailjwt/test/test_jwt.py (hunk 2 above)
Comment (danking) [reply hailjwt/test/test_jwt.py]: corrupted key means a key that's too short?

File: hailjwt/hailjwt/hailjwt.py (hunk 4 above)
Comment (danking) [reply hailjwt/hailjwt/hailjwt.py]: I think the secret key should generally come from outside the application, right?

Comment (cseed) [reply hailjwt/hailjwt/hailjwt.py]: I'm suggesting we have a utility for generating that key.

File: hailjwt/test/test_jwt.py (hunk 2 above)
Comment (cseed) [reply hailjwt/test/test_jwt.py]: I'd like to see tests that verify:
 - an attempt to modify the cypher text is detected,
 - if the cipher text was signed by another key (e.g. one generated by an attacker), it is detected.

File: hailjwt/test/test_jwt.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+def test_round_trip():
+    c = hj.JWTClient(secrets.token_bytes(256))
+    json = {'hello': 'world'}
+    assert c.decode(c.encode(json)) == json
+
+
+def test_fewer_than_256_bits_is_error():
+    try:
+        hj.JWTClient(secrets.token_bytes(255))
```
Comment (cseed) [inline hailjwt/test/test_jwt.py]: That's 2040 bits.

File: hailjwt/hailjwt/hailjwt.py (hunk 4 above)
Comment (danking) [reply hailjwt/hailjwt/hailjwt.py]: done

File: hailjwt/test/test_jwt.py (hunk 2 above)
Comment (danking) [reply hailjwt/test/test_jwt.py]: done

**Comment by danking** [comment]
This should start passing when "add batch docker dependency .." https://github.com/hail-is/hail/pull/5655/files merges.

**Comment by danking** [comment]
@jigold cotton & alex are out, so reassigning to you

**Comment by danking** [comment]
bump @jigold I need this to go in for my batch PR.

**Comment by jigold** [comment]
How did you fix it to use the right version of python? Otherwise, everything looks good to me.

**Comment by danking** [comment]
@jigold I changed instances of `python` to `python3`. cotton's conda change necessitated this

**Pull request merged**